src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalFileProvider.cs

```diff
@@ -297,7 +300,7 @@ public IFileInfo GetFileInfo(string subpath)
         /// <paramref name="subpath" /> is absolute, if the directory does not exist, or <paramref name="subpath" /> has invalid
         /// characters.
         /// </returns>
-        public IDirectoryContents GetDirectoryContents(string subpath)
+        public IDirectoryContents GetDirectoryContents(string? subpath)
```

Comment:
while technically `subpath` won't throw on null I have very mixed feelings about this being nullable since that might suggest that it will return all contents rather than always returning nothing... IMO this should be non-null... @stephentoub thoughts?

Reply:
I would second this. If this is non-nullable, than `IFileProvider.GetDirectoryContents subpath` also can be marked as non-nullable

Reply:
Made `subpath` non-nullable